plugins/modules/secrets_provider_vault.py

```diff
@@ -0,0 +1,246 @@
+    type: dict
+    suboptions:
+      ca_cert_file:
+        description:
+          - Path to the certificate file of the trusted certificate authority.
+        type: str
+      client_cert_file:
+        description:
+          - Path to the client certificate file.
+        type: str
+      client_key_file:
+        description:
+          - Path to the client key file.
+        type: str
+      common_name:
+        description:
+          - Common name of the Vault server.
+        type: str
+
+seealso:
+  - module: secrets_provider_env
+  - module: secrets_provider_info
+  - module: secret
+'''
+
+EXAMPLES = '''
+- name: Create a vault secrets provider
+  sensu.sensu_go.secrets_provider_vault:
+    name: my-vault
+    address: https://my-vault.com
+    token: VAULT_TOKEN
+    version: v1
+
+- name: Delete a vault secrets provider
+  sensu.sensu_go.secrets_provider_vault:
+    name: my-vault
+    state: absent
+'''
+
+RETURN = '''
+object:
+    description: object representing Sensu vault secrets provider
+    returned: success
+    type: dict
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.sensu.sensu_go.plugins.module_utils import (
+    arguments, errors, utils
+)
+
+API_GROUP = "enterprise"
+API_VERSION = "secrets/v1"
+
+
+def do_differ(current, desired):
+    if utils.do_differ_v1(current, desired, "client"):
+        return True
+
+    current_client = current["spec"]["client"]
+    desired_client = desired["spec"]["client"]
+    # Sensu Go API returns 'agent_address' field in the client spec,
+    # but this field is not meant to be set via the providers API.
+    if utils.do_differ(current_client, desired_client, "agent_address", "tls"):
+        return True
+    # Sensu Go API returns 'insecure' and 'tls_server_name' fields in the
+    # tls spec. We ignore them, as they are not meant to be set via the
+    # providers API.
+    return utils.do_differ(current_client["tls"], desired_client.get("tls") or {},
+                           "insecure", "tls_server_name")
+
+
+def _format_seconds(seconds):
+    # Sensu API returns the configured timeout as a string, for instance
+    # 30 -> '30s', 60-> '1m0s', 3600 -> '1h0m0s'.
+    if seconds < 60:
+        return "{0}s".format(seconds)
+    if seconds < 3600:
+        m, s = divmod(seconds, 60)
+        return "{0}m{1}s".format(m, s)
+    h, r = divmod(seconds, 3600)
+    m, s = divmod(r, 60)
+    return "{0}h{1}m{2}s".format(h, m, s)
+
+
+def build_vault_provider_spec(params):
+    if params["state"] == "absent":
+        return {}
+
+    client = arguments.get_spec_payload(
+        params, "address", "token", "version", "max_retries")
+    if params.get("tls"):
+        client["tls"] = arguments.get_renamed_spec_payload(params["tls"], dict(
+            ca_cert_file="ca_cert",
+            client_cert_file="client_cert",
+            client_key_file="client_key",
+            common_name="cname",
```

Comment:
I am not sure about this renaming here. I wholeheartedly agree that upstream parameter names are less that ideal and do not convey all the information they should. But it still feels to me that keeping familiarity with the upstream API might be more important here. Using upstream docs for reference in vast majority of cases is something our users really like.

Reply:
I was actually not sure whether this renaming would be a good thing or a bad one. Thanks for your input, we're now aligned with the upstream.